ccm_web/server/src/canvas/canvas.service.ts

```diff
@@ -14,47 +14,62 @@ import { User } from '../user/user.model'
 import { CanvasConfig } from '../config'
 import { DatabaseError } from '../errors'
 import baseLogger from '../logger'
-import { IncomingHttpHeaders } from 'http'
 
 const logger = baseLogger.child({ filePath: __filename })
 
 type SupportedAPIEndpoint = '/api/v1/' | '/api/graphql/'
 
-interface IncomingRateLimitedCanvasHttpHeaders extends IncomingHttpHeaders {
-  'x-rate-limit-remaining'?: string
-  'x-request-cost'?: string
-}
-
 const requestorOptions: GotOptions = {
   retry: {
-    limit: 2,
+    limit: 4,
     methods: ['POST', 'GET', 'PUT', 'DELETE'],
-    statusCodes: got.defaults.options.retry.statusCodes.concat([403])
+    statusCodes: got.defaults.options.retry.statusCodes.concat([403]),
+    calculateDelay: ({ attemptCount, retryOptions, error, computedValue }) => {
+      const isStrangeError = !(error instanceof HTTPError) && error.code === undefined
```

Comment:
Do you still think the `isStrangeError` handling is necessary? I'm not sure I ever saw it be `true` after I implemented it, maybe the incorrect `headers` accessing explains all the error cases you were seeing, I'm not sure.

Reply:
I wasn't sure whether you saw that case being handled or not, so I left that code in place.  In my testing, I've not seen any value for that except `false`.  If you and @pushyamig don't see that case, either, then I will remove it.

Reply:
I didn't see it in my testing again either. I'm gonna say remove it. Maybe I was getting confused, and the issue was wholly a `headers` one.